src/AutoRest.CSharp/Common/Output/Models/Types/SignatureType.cs

```diff
@@ -0,0 +1,318 @@
+                }
+            }
+            return overloadMethods;
+        }
+
+        private MethodBodyStatement BuildOverloadMethodBody(OverloadMethodSignature overloadMethodSignature)
+            => Return(new InvokeInstanceMethodExpression(null, overloadMethodSignature.MethodSignature.Name, BuildOverloadMethodParameters(overloadMethodSignature), null, overloadMethodSignature.PreviousMethodSignature.Modifiers.HasFlag(MethodSignatureModifiers.Async)));
+
+        private IReadOnlyList<ValueExpression> BuildOverloadMethodParameters(OverloadMethodSignature overloadMethodSignature)
+        {
+            var parameters = new List<ValueExpression>();
+            var set = overloadMethodSignature.MissingParameters.ToHashSet(Parameter.TypeAndNameEqualityComparer);
+            foreach (var parameter in overloadMethodSignature.MethodSignature.Parameters)
+            {
+                if (set.Contains(parameter))
+                {
+                    parameters.Add(new PositionalParameterReference(parameter.Name, Default.CastTo(parameter.Type)));
+                    continue;
+                }
+                parameters.Add(new PositionalParameterReference(parameter));
+            }
+            return parameters;
+        }
+
+        private IReadOnlySet<MethodSignature>? _methodsToSkip;
+        public IReadOnlySet<MethodSignature> MethodsToSkip => _methodsToSkip ??= EnsureMethodsToSkip();
+        private IReadOnlySet<MethodSignature> EnsureMethodsToSkip()
+        {
+            if (_customization is null)
+            {
+                return new HashSet<MethodSignature>();
+            }
+            return Methods.Intersect(_customization.Methods, MethodSignature.ParameterAndReturnTypeEqualityComparer).ToHashSet(MethodSignature.ParameterAndReturnTypeEqualityComparer);
+        }
+
+        private bool TryGetPreviousMethodWithLessOptionalParameters(IList<MethodSignature> currentMethods, MethodSignature previousMethod, [NotNullWhen(true)] out MethodSignature? currentMethodToCall, [NotNullWhen(true)] out IReadOnlyList<Parameter>? missingParameters)
+        {
+            foreach (var item in currentMethods)
+            {
+                if (item.Parameters.Count <= previousMethod.Parameters.Count)
+                {
+                    continue;
+                }
+
+                if (!CurrentContainAllPreviousParameters(previousMethod, item))
+                {
+                    continue;
+                }
+
+                // We can't use CsharpType.Equals here because they could have different implementations from different versions
+                if (previousMethod.ReturnType is null && item.ReturnType is not null)
+                {
+                    continue;
+                }
+                if (previousMethod.ReturnType is not null && !previousMethod.ReturnType.EqualsByName(item.ReturnType))
+                {
+                    continue;
+                }
+
```

Comment:
could we add some descriptions to these branches?

Reply:
I think these code branches should explain themselves, otherwise we should have a better name for the methods.

Reply:
well I mean adding some description as "why" not just explaining what condition this branch is checking, which just works like a "document"